kraken2: init at 2.1.5

cardealerT opened the pull request:
## Description of Changes
Added kraken2 taxonomic sequence classification system to by-name/kr

homepage: https://ccb.jhu.edu/software/kraken2/

github: https://github.com/DerrickWood/kraken2

Also added cardealerT and A1egator to maintainers list

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MarcFontaine committed:
libacars: init at 2.2.0

pks-t committed:
opensmtpd.table-sqlite: init at 1.0.1

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-sqlite, which is
one of these new standalone filters to read login info from an sqlite
database.

pks-t committed:
nixos/opensmtpd: adapt to changed filter location

The location of filters for OpenSMTPD has been changed to
"libexec/smtpd". Adapt the service accordingly.

pks-t committed:
opensmtpd.table-ldap: init at 1.0

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-ldap, which is one
of these new standalone filters to read login info from an LDAP server.

pks-t committed:
opensmtpd.table-mysql: init at 1.2.1

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-mysql, which is
one of these new standalone filters to read login info from a MySQL or
MariaDB database.

pks-t committed:
opensmtpd.table-passwd: init at 1.0.2

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-passwd, which is
one of these new standalone filters to read login info from "passwd"
files.

pks-t committed:
opensmtpd.table-postgres: init at 1.1.1

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-postgres, which
is one of these new standalone filters to read login info from a
Postgres database.

pks-t committed:
opensmtpd.table-redis: init at 1.0.1

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-redis, which is
one of these new standalone filters to read login info from a Redis
instance.

Building table-redis does not work out of the box: while the project
correctly includes the hiredis include directory, the header itself is
contained in a "hiredis/" subdirectory. Add a patch to fix this.

pks-t committed:
opensmtpd.table-socketmap: init at 1.1.1

The opensmtpd-extra package has been deprecated in favor of standalone
repositories for each of the filters. Introduce table-socketmap, which
is one of these new standalone filters to read login info from a Unix
socket.

Jo-Blade committed:
python3Packages.gotify: init at 0.6.0

pks-t committed:
opensmtpd-extras: drop in favor of standalone tables

Upstream has archived the OpenSMTPD-extras repository with 82c49af (add
deprecation notice, 2024-10-22). Instead, it has created a set of
repositories, each containing one of the tables. The old tables aren't
compatible anymore with recent OpenSMTPD versions due to changed
protocols. Other extras, like the queues, don't have an alternative to
the best of my knowledge.

Furthermore, opensmtpd-extras doesn't even build anymore, furthermore
indicating that there aren't any users of this package. Remove the
"opensmtpd-extras" package and raise errors pointing to the new
standalone tables.

Add a release note.

r-ryantm committed:
python312Packages.pypugjs: 5.10.1 -> 5.12.0

Commit:
conduktor: 2.15.1 -> 2.24.9

Diff: https://github.com/conduktor/builds/compare/v2.15.1...v2.24.9

Changelog: https://www.conduktor.io/changelog/#2.24.9

Commit:
contour: 0.5.1.7247 -> 0.6.1.7494

Diff: https://github.com/contour-terminal/contour/compare/v0.5.1.7247...v0.6.1.7494

Changelog: https://github.com/contour-terminal/contour/raw/v25.05pre-git/Changelog.md

Commit:
crc: 2.36.0 -> 2.49.0

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+
+  src = fetchFromGitHub {
+    owner = "DerrickWood";
+    repo = pname;
```
```suggestion
    repo = "kraken";
```
Do not rely on repo and program names match, it can change.

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+  src = fetchFromGitHub {
+    owner = "DerrickWood";
+    repo = pname;
+    rev = "v2.14"; # typo of current kraken2 build. version of source and software are different.
```
Derrick has already released 2.1.5 with a correct tag, so you can use `tag = "v${finalAttrs.version}"` (`rev` is reserved for non-tag refs and specific commits)

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+{ lib, stdenv, fetchFromGitHub, rsync, perl, wget, gnused, findutils, libgcc, llvmPackages, zlib }:
+
+stdenv.mkDerivation rec {
```
Speaking of `finalAttrs`:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+    platforms = platforms.unix;
+    maintainers = [ maintainers.cardealerT maintainers.A1egator];
+  };
+}
```
```suggestion
})
```

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+    owner = "DerrickWood";
+    repo = pname;
+    rev = "v2.14"; # typo of current kraken2 build. version of source and software are different.
+    sha256 = "sha256-PneYibG7/u1M32f1irvBwUWyrE4gBL3evVCSj94YGQM=";
```
```suggestion
    hash = "sha256-PneYibG7/u1M32f1irvBwUWyrE4gBL3evVCSj94YGQM=";
```
Ah yes, just use `hash` everywhere

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+    sha256 = "sha256-PneYibG7/u1M32f1irvBwUWyrE4gBL3evVCSj94YGQM=";
+  };
+
+  buildInputs = [
+    rsync
+    perl
+    wget
+    gnused
+    findutils
+    libgcc
+    llvmPackages.openmp
+    zlib
+  ];
```
`buildInputs` is for libraries to link against or source from (if it's an interpreted program), for stuff that is supposed to get executed use `nativeBuildInputs` (also `gnused`, `findutils` and `libgcc` are already parts of `stdenv`, adding them here is redundant)

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+    patchShebangs ./install_kraken2.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ./install_kraken2.sh $out/bin
+  '';
```
The repo already has a Makefile, have you tried building with it instead of the script? Because, since you haven't overriden `buildPhase`, it's gonna be executed. You might wanna specify smth like this instead: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/terminal-emulators/st/default.nix#L67.

Also, when overriding the entire phase, make sure to leave `runHook preInstall` and `runHook postInstall` in the beginning and in the end, without them it'll be impossible to add extra build steps.

acid-bong commented on pkgs/by-name/kr/kraken2/package.nix:
```diff
@@ -0,0 +1,45 @@
+    ./install_kraken2.sh $out/bin
+  '';
+
+  meta = with lib; {
+    description = ''
+      The second version of the Kraken taxonomic sequence classification system, 
+      using exact k-mer matches to achieve high accuracy and fast classification speeds.
+    '';
+    homepage = "https://ccb.jhu.edu/software/kraken2/";
+    license = licenses.mit;
+    platforms = platforms.unix;
+    maintainers = [ maintainers.cardealerT maintainers.A1egator];
+  };
```
```suggestion
  meta = {
    description = ''
      The second version of the Kraken taxonomic sequence classification system, 
      using exact k-mer matches to achieve high accuracy and fast classification speeds.
    '';
    homepage = "https://ccb.jhu.edu/software/kraken2/";
    changelog = "https://github.com/DerrickWood/kraken2/releases/tag/v${finalAttrs.version}";
    license = lib.licenses.mit;
    platforms = platforms.unix;
    maintainers = with lib.maintainers; [
      cardealerT
      A1egator
    ];
  };
```
`with lib;` increases eval time, which is no good, use `with` for smaller namespaces, like `lib.maintainers`

acid-bong reviewed (changes requested):
Phew, that's a lot.

Also format with `nixfmt-rfc-style`.

Commit:
ddosify: 1.0.6 -> 2.6.0

acid-bong reviewed (changes requested):
The contents are good now. Now squash the last two commits[^1] and edit them according to the convention (same footer, actually)

[^1]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

CherryKitten committed:
gotosocial: move to codeberg

doronbehar committed:
sshd-openpgp-auth: 0.3.0 -> 0.3.1

doronbehar committed:
ssh-openpgp-auth: 0.2.2 -> 0.2.3

FliegendeWurst committed:
xastir: fix cross, enable strictDeps, update meta, modernize

sarahec committed:
code-cursor: 0.48.9 -> 0.49.6

pks-t committed:
opensmtpd-filter-rspamd: migrate to by-name

pks-t committed:
opensmtpd: migrate to by-name

Commit:
gkrellm: 2.3.11 -> 2.4.0

numinit committed:
ponyc, pony-corral: move to by-name

iynaix committed:
mpvScripts.eisa01.undoredo: init at 0-unstable-2023-11-25 (#401930)

wegank committed:
python312Packages.pypugjs: modernize

luftmensch-luftmensch committed:
mpvScripts.eisa01.simplebookmark: init at 0-unstable-2023-09-22 (#402041)

GaetanLepage committed:
vimPlugins.bitbake-vim: remove updateScript

wegank committed:
Revert "file: patch to fix misclassification of some zip files"

wegank committed:
Revert "file: patch to fix misclassification of some zip files" (#402311)

drupol committed:
vscode-extensions.golang.go: 0.47.1 -> 0.47.2 (#402264)

drupol committed:
texmaker: 6.0.0 -> 6.0.1 (#402178)

drupol committed:
vscode-extensions: update (#402141)

wegank committed:
linuxPackages.nullfs: 0.17 -> 0.18 (#395412)

doronbehar committed:
Revert "file: patch to fix misclassification of some zip files"

doronbehar committed:
Revert "file: patch to fix misclassification of some zip files" (#402316)

wolfgangwalther committed:
youtrack: 2025.1.67057 -> 2025.1.71685 (#397609)

wolfgangwalther committed:
b3sum: 1.8.0 -> 1.8.2 (#397757)

FliegendeWurst committed:
molecule: 25.3.1 -> 25.4.0 (#401327)

wolfgangwalther committed:
tomcat9: 9.0.102 -> 9.0.104 (#399044)

wegank committed:
python312Packages.hikari-crescent: 1.1.0 -> 1.2.0 (#390921)

doronbehar committed:
musescore: 4.5.1 -> 4.5.2 (#401795)

wegank committed:
python312Packages.karton-core: 5.6.0 -> 5.6.1 (#395682)

4evy committed:
starship: 1.22.1 -> 1.23.0

Changelog: https://github.com/starship/starship/releases/tag/v1.23.0
Diff: https://github.com/starship/starship/compare/v1.22.1...v1.23.0

drupol committed:
ldmud: move to pkgs/by-name (#402169)

wolfgangwalther committed:
vscode-extensions.ms-dotnettools.csdevkit: 1.18.23 -> 1.18.25 (#399180)

flokli committed:
secretscanner: build with vectorscan instead of hyperscan

See https://github.com/NixOS/nixpkgs/issues/310772 for context.

This brings us aarch64-linux support for free.

wolfgangwalther committed:
moodle: 4.5.3 -> 5.0 (#398571)

7c6f434c committed:
bftpd: 6.2 -> 6.3 (#401920)

SuperSandro2000 committed:
youtube-music: 3.8.1 -> 3.9.0 (#402248)

siraben committed:
chez: pin to clang-17 on aarch64 darwin (#393687)

TomaSajt committed:
trickster: move to pkgs/by-name

khaneliman committed:
vimPlugins: update on 2025-04-26 (#402305)

khaneliman committed:
luaPackages: update on 2025-04-27

wolfgangwalther committed:
libpqxx: 7.10.0 -> 7.10.1 (#396024)

wolfgangwalther committed:
video-compare: 20250223 -> 20250420 (#399735)

wolfgangwalther committed:
jmol: 16.3.13 -> 16.3.17 (#399373)

wolfgangwalther committed:
trickster: move to pkgs/by-name (#402331)

sarahec committed:
code-cursor: use chromium's inbuilt x11wayland renderer by defaullt

wolfgangwalther committed:
lego: 4.22.2 -> 4.23.1 (#399302)

nbdd0121 committed:
python-matter-server: update bundled certificates

Notably this picks up certificates from Aqara.

markuskowa committed:
python312Packages.m2crypto: 0.44.0 -> 0.45.0 (#400875)

markuskowa committed:
stylua: 2.0.2 -> 2.1.0 (#400603)

wolfgangwalther committed:
orchard: 0.31.0 -> 0.32.2 (#400508)

wolfgangwalther committed:
amazon-ssm-agent: 3.3.1957.0 -> 3.3.2299.0 (#400156)

getchoo committed:
skim: 0.16.1 -> 0.16.2 (#402133)

trofi committed:
bombono: bin to boost183 to fix the build

Without the change `bombono` build fails on `master` as
https://hydra.nixos.org/build/294719057:

    In file included from src/mbase/project/_extlibs_.h:61,
                     from src/mgui/_extlibs_.h:31,
                     from src/mgui/_pc_.h:29,
                     from src/mgui/dialog.cpp:22:
    src/mlib/sdk/bfs.h:12:10: fatal error: boost/filesystem/convenience.hpp: No such file or directory
       12 | #include <boost/filesystem/convenience.hpp> // fs::create_directories()
          |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

wolfgangwalther committed:
boinc: 8.0.4 -> 8.2.1 (#398133)

wolfgangwalther committed:
anchor: 0.31.0 -> 0.31.1 (#400211)

drupol committed:
dprint-plugins: add installCheckPhase (#402056)

drupol committed:
starship: 1.22.1 -> 1.23.0 (#402324)

wolfgangwalther committed:
sby: 0.51 -> 0.52 (#398466)

wolfgangwalther committed:
tt-rss : Cleanup, better tests, mark plugins as broken (#367736)

drupol committed:
vscode-extensions.streetsidesoftware.code-spell-checker-french: init at 0.4.3 (#402250)

wolfgangwalther committed:
thanos: 0.37.2 -> 0.38.0 (#397268)

drupol committed:
vscode-extensions.github.copilot: 1.303.0 -> 1.309.0; vscode-extensions.github.copilot-chat: 0.26.2025040204 -> 0.26.7 (#402022)

happysalada committed:
stalwart-mail: darwin cleanup and skip unreliable test (#402289)

mweinelt committed:
evcc: 0.203.3 -> 0.203.4

https://github.com/evcc-io/evcc/releases/tag/0.203.4

sarahec committed:
code-cursor: disable automatic updates

mweinelt committed:
python-matter-server: update bundled certificates (#402337)

OPNA2608 committed:
{ayatana-indicator*,lomiri.lomiri-indicator-network}: Add corresponding Lomiri indicator test to passthru.tests (#397807)

drupol committed:
starship: cleanup

r-ryantm committed:
nuclei-templates: 10.1.7 -> 10.2.0

nixpkgs-merge-bot[bot] committed:
msbuild-structured-log-viewer: 2.2.465 -> 2.2.476 (#401496)

Co-authored-by: js6pak <js6pak@users.noreply.github.com>

pbsds committed:
cutentr: init at 0.3.3 (#394476)

r-ryantm committed:
python312Packages.pyghmi: 1.5.77 -> 1.6.0

dywedir committed:
oxipng: 9.1.4 -> 9.1.5 (#402156)

r-ryantm committed:
chirpstack-rest-api: 4.11.0 -> 4.12.0

r-ryantm committed:
chirpstack-concentratord: 4.4.7 -> 4.4.8

r-ryantm committed:
chirpstack-mqtt-forwarder: 4.3.1 -> 4.3.2

happysalada committed:
mise: 2025.4.1 -> 2025.4.11 (#400397)

r-ryantm committed:
chirpstack-udp-forwarder: 4.1.8 -> 4.1.10

fpletz committed:
librewolf-unwrapped: 137.0.1-1 -> 137.0.2-1 (#400213)

wegank committed:
easycrypt: 2025.02 -> 2025.03 (#393192)

azahi committed:
mold: 2.37.1 -> 2.38.0 (#402255)

wegank committed:
fluidd: 1.33.0 -> 1.34.1 (#396925)

wegank committed:
opentelemetry-collector-builder: 0.122.1 -> 0.124.0 (#395056)

wegank committed:
python312Packages.localstack-client: 2.7 -> 2.10 (#395842)

fpletz committed:
wireshark: 4.4.5 -> 4.4.6 (#399365)

wegank committed:
yquake2: 8.41 -> 8.50 (#396696)

wegank committed:
python312Packages.gaphas: 5.0.0 -> 5.0.3 (#397361)

isabelroses committed:
sof-firmware: clean up package; provide update script (#401442)

wegank committed:
python312Packages.azure-mgmt-postgresqlflexibleservers: 1.1.0b1 -> 1.1.0 (#395078)

FliegendeWurst committed:
vintagestory: 1.20.7 -> 1.20.9 (#401528)

wegank committed:
ocamlPackages.tls: 2.0.0 -> 2.0.1 (#398577)

wegank committed:
python312Packages.configshell-fb: 2.0.0 -> 2.0.2 (#398330)

wegank committed:
ocamlPackages.qcheck-multicoretests-util: 0.7 -> 0.8 (#397788)

commiterate committed:
gitlab-runner: Use substituteInPlace for patching and stop referencing src attributes

wegank committed:
ocamlPackages.digestif: 1.2.0 -> 1.3.0 (#398572)

wegank committed:
ocamlPackages.ca-certs: 1.0.0 -> 1.0.1 (#398955)

wegank committed:
ocamlPackages.x509: 1.0.5 -> 1.0.6 (#398953)

wegank committed:
ocamlPackages.tcpip: 9.0.0 -> 9.0.1 (#398954)

trofi committed:
xsokoban: fix build agaist gcc-14

It's a really old program. Let's disable most implicit
declarations to restore the build.

wegank committed:
python312Packages.ppft: 1.7.6.9 -> 1.7.7 (#400580)

wegank committed:
mysql_jdbc: 9.2.0 -> 9.3.0 (#399987)

wegank committed:
survex: 1.4.16 -> 1.4.17 (#397541)

GaetanLepage committed:
vimPlugins.bitbake-vim: remove updateScript (#402312)

wegank committed:
python312Packages.pytapo: 3.3.42 -> 3.3.44 (#400301)

numinit committed:
androidenv: support linking "latest" version of all plugins

As noted in #379534, the NDK linking broke during refactoring. Add a test
for that, and support linking the latest for all platform plugins that
take at least one version.

wegank committed:
autobrr: 1.60.0 -> 1.61.0 (#400695)

wegank committed:
buildah-unwrapped: 1.39.4 -> 1.40.0 (#400629)

GaetanLepage committed:
gitlab-runner: Use `substituteInPlace` for patching and stop referencing `src` attributes (#402379)

GaetanLepage committed:
luaPackages: update on 2025-04-27 (#402333)

r-ryantm committed:
roddhjav-apparmor-rules: 0-unstable-2025-04-16 -> 0-unstable-2025-04-25

khaneliman committed:
vimPlugins.copilot-lsp: init at 2025-04-27

r-ryantm committed:
python312Packages.uiprotect: 7.5.4 -> 7.5.5 (#402328)

https://github.com/uilibs/uiprotect/blob/v7.5.5/CHANGELOG.md

mweinelt committed:
python3Packages.gotify: init at 0.6.0 (#382608)

* maintainers: add joblade

* python3Packages.gotify: init at 0.6.0

numinit committed:
androidenv: update docs to explain the latest version

wegank committed:
python312Packages.django-ninja: 1.4.0 -> 1.4.1 (#399733)

r-ryantm committed:
python312Packages.onedrive-personal-sdk: 0.0.13 -> 0.0.14

mweinelt committed:
evcc: 0.203.3 -> 0.203.4 (#402390)

GaetanLepage committed:
vimPlugins.copilot-lsp: init at 2025-04-27 (#402386)

GaetanLepage committed:
code-cursor: 0.48.9 -> 0.49.6 (#401864)

natsukium committed:
python312Packages.ale-py: 0.10.2 -> 0.11.0 (#402003)

natsukium committed:
python312Packages.pypugjs: 5.10.1 -> 5.12.0 (#394934)

natsukium committed:
python312Packages.yalexs-ble: 2.5.7 -> 2.6.0 (#402325)

r-ryantm committed:
wxmaxima: 25.01.0 -> 25.04.0

r-ryantm committed:
rip2: 0.9.3 -> 0.9.4

natsukium committed:
python312Packages.securesystemslib: 1.2.0 -> 1.3.0 (#401401)

natsukium committed:
python312Packages.symengine: 0.14.0 -> 0.14.1 (#400814)

nixpkgs-merge-bot[bot] committed:
ethercat: 1.6.3 -> 1.6.4 (#401799)

Co-authored-by: stv0g <post@steffenvogel.de>

nixpkgs-merge-bot[bot] committed:
chirpstack-rest-api: 4.11.0 -> 4.12.0 (#402365)

Co-authored-by: stv0g <post@steffenvogel.de>

nixpkgs-merge-bot[bot] committed:
chirpstack-concentratord: 4.4.7 -> 4.4.8 (#402366)

Co-authored-by: stv0g <post@steffenvogel.de>

nixpkgs-merge-bot[bot] committed:
chirpstack-mqtt-forwarder: 4.3.1 -> 4.3.2 (#402367)

Co-authored-by: stv0g <post@steffenvogel.de>

nixpkgs-merge-bot[bot] committed:
chirpstack-udp-forwarder: 4.1.8 -> 4.1.10 (#402370)

Co-authored-by: stv0g <post@steffenvogel.de>

drupol committed:
starship: cleanup (#402349)

nixpkgs-merge-bot[bot] committed:
borgmatic: 2.0.2 -> 2.0.4 (#401040)

Co-authored-by: x123 <x123@users.noreply.github.com>

r-ryantm committed:
readarr: 0.4.14.2782 -> 0.4.15.2787

drupol committed:
gitify: 6.1.0 -> 6.3.0 (#401939)

drupol committed:
gcompris: 4.3 -> 25.0 (#401292)

drupol committed:
cherry-studio: 1.2.4 -> 1.2.7 (#401788)

drupol committed:
gigedit: 1.1.1 -> 1.2.1 (#401936)

drupol committed:
fuc: 2.2.0 -> 3.0.1 (#400698)

drupol committed:
butterfly: 2.2.4 -> 2.3.0 (#400635)

drupol committed:
dtbloader: 1.2.2 -> 1.4.0 (#400658)

drupol committed:
dpdk: 24.07 -> 25.03 (#400621)

drupol committed:
detox: 1.4.5 -> 2.0.0 (#400587)

drupol committed:
dbgate: 6.2.0 -> 6.3.3 (#400498)

drupol committed:
cups-bjnp: 1.2.2 -> 2.0.3 (#400475)

drupol committed:
dump1090: 9.0 -> 10.0.1 (#400660)

drupol committed:
e2tools: 0.1.0 -> 0.1.2 (#400664)

drupol committed:
bitcomet: 2.12.1 -> 2.13.1 (#400441)

drupol committed:
saber: 0.25.4 -> 0.25.5 (#400445)

drupol committed:
simple-live-app: 1.7.6 -> 1.7.7 (#400449)

drupol committed:
proxypin: 1.1.7 -> 1.1.8 (#400453)

drupol committed:
chatmcp: 0.0.21-alpha -> 0.0.26-alpha (#400454)

drupol committed:
diylc: 4.18.0 -> 5.2.0 (#400595)

drupol committed:
doitlive: 5.0.0 -> 5.1.0 (#400614)

drupol committed:
convertall: 0.8.0 -> 1.0.1 (#400204)

drupol committed:
CuboCore: 4.5.0 -> 5.0.0 (#400220)

drupol committed:
croaring: 2.0.3 -> 4.3.1 (#400341)

drupol committed:
csvkit: 2.0.1 -> 2.1.0; python313Packages.agate: 1.11.0 -> 1.13.0 (#400348)

drupol committed:
conan: 2.9.1 -> 2.15.1 (#400174)

drupol committed:
confluent-cli: 4.16.0 -> 4.25.0 (#400197)

r-ryantm committed:
syft: 1.22.0 -> 1.23.1

drupol committed:
libcef: refactor (#397452)

drupol committed:
flclash: 0.8.80 -> 0.8.82 (#397179)

drupol committed:
windsend: init at 1.5.1 (#396431)

drupol committed:
fvm: init at 3.2.1 (#396140)

drupol committed:
kazumi: 1.5.6 -> 1.6.5 (#394274)

drupol committed:
venera: 1.2.4 -> 1.4.0 (#388162)

drupol committed:
surface-control: 0.4.3-2 -> 0.4.7-1 (#386296)

drupol committed:
tiny-rdm: init at 1.2.3 (#398390)

drupol committed:
reqable: init at 2.33.8 (#398468)

drupol committed:
cromite: 134.0.6998.108 -> 135.0.7049.100 (#396081)

drupol committed:
alisthelper: init at 0.2.0-unstable-2025-01-04 (#396137)

drupol committed:
xmind: 8-update9 -> 25.01.01061-202501070800 (#393425)

drupol committed:
yo: remove package-lock.json (#397568)

wolfgangwalther committed:
sitespeed-io: 37.3.2 -> 37.4.2 (#398870)

doronbehar committed:
ssh-openpgp-auth: 0.2.2 -> 0.2.3 (#401398)

7c6f434c committed:
xsokoban: fix build agaist gcc-14 (#402377)

drupol committed:
Revert "dpdk: 24.07 -> 25.03"

drupol committed:
Revert "dpdk: 24.07 -> 25.03" (#402456)

acid-bong reviewed (changes requested):
rename ALL commits according to the conventions

numinit committed:
ponyc: 0.58.6 -> 0.59.0 (#396553)

drupol committed:
rexi: init at 1.2.0 (#398469)

r-vdp committed:
age-plugin-se: build on linux (#401265)

drupol committed:
slint-viewer: init at 1.10.0 (#394944)

drupol committed:
hanken-grotesk: init at 0-unstable-2024-01-30 (#397215)

drupol committed:
chrome-pak-customizer: init at 2.0-unstable-2021-06-24 (#401482)

r-ryantm committed:
mycelium: 0.5.7 -> 0.6.0

GaetanLepage committed:
ocserv: add otp support (#395801)

Mic92 committed:
signal-desktop-bin(x86_64-linux; aarch64-linux; darwin): 7.47.0 -> 7.52.0; signal-desktop-bin: replace apple emoji sheets  (#400945)

FliegendeWurst committed:
libacars: init at 2.2.0 (#402348)

fabaff committed:
nuclei-templates: 10.1.7 -> 10.2.0 (#402355)

happysalada committed:
pythonPackages.polars: 1.21.0 -> 1.27.1 (#401298)

fabaff committed:
ldeep: unbreak by relaxing termcolor (#402068)

etu committed:
testssl: 3.0.9 -> 3.2.0 (#401112)

Commit:
reflection-cpp: init at 0.2.0

nixpkgs-merge-bot[bot] committed:
readarr: 0.4.14.2782 -> 0.4.15.2787 (#402434)

Co-authored-by: jocelynthode <jocelynthode@users.noreply.github.com>

fabaff committed:
forbidden: refactor

GaetanLepage committed:
python312Packages.batchspawner: fix

GaetanLepage committed:
python312Packages.jupyterhub: cleanup

r-vdp committed:
nixos/mycelium: pull in network-online.target

The test was failing because it waits for network-online.target which
was not part of the transaction.

fabaff committed:
forbidden: 12.6 -> 13.2 (#400696)

fabaff committed:
secretscanner: build with vectorscan instead of hyperscan (#402329)

GaetanLepage committed:
python312Packages.jupyterhub: 5.2.1 -> 5.3.0

Diff: https://github.com/jupyterhub/jupyterhub/compare/refs/tags/5.2.1...refs/tags/5.3.0

Changelog: https://github.com/jupyterhub/jupyterhub/blob/5.3.0/docs/source/reference/changelog.md

doronbehar committed:
wxmaxima: 25.01.0 -> 25.04.0 (#402409)

Commit:
dnsviz: move to pkgs/by-name

nixpkgs-merge-bot[bot] committed:
your_spotify: 1.12.0 -> 1.13.1 (#402213)

Co-authored-by: PatrickDaG <PatrickDaG@users.noreply.github.com>

katexochen committed:
libnvidia-container: 1.17.2 -> 1.17.6

Signed-off-by: Paul Meyer <katexochen0@gmail.com>

r-ryantm committed:
zed-editor: 0.183.10 -> 0.183.11

dotlambda committed:
python312Packages.onedrive-personal-sdk: 0.0.13 -> 0.0.14 (#402393)

teto committed:
yaziPlugins.rsync: init at 0-unstable-2025-04-12

K900 committed:
Revert "slint-viewer: init at 1.10.0"

K900 committed:
Revert "slint-viewer: init at 1.10.0" (#402486)

drupol committed:
gkrellm: 2.3.11 -> 2.4.0 (#401944)

katexochen committed:
libnvidia-container: renew

- use finalAttrs
- use tag in fetchFromGithub
- no meta = with lib;

Signed-off-by: Paul Meyer <katexochen0@gmail.com>

r-vdp committed:
mycelium: 0.5.7 -> 0.6.0 (#402466)

drupol committed:
crc: 2.36.0 -> 2.49.0 (#400339)

drupol committed:
contour: 0.5.1.7247 -> 0.6.1.7494 (#400200)

drupol committed:
ddosify: 1.0.6 -> 2.6.0 (#400560)

nixpkgs-merge-bot[bot] committed:
rip2: 0.9.3 -> 0.9.4 (#402411)

Co-authored-by: matthiasbeyer <mail@beyermatthias.de>

drupol committed:
conduktor: 2.15.1 -> 2.24.9 (#400188)

drupol committed:
dnsviz: 0.9.4 -> 0.11.1 (#400601)

natsukium committed:
python312Packages.jupyterhub: 5.2.1 -> 5.3.0, cleanup (#402473)

nixpkgs-merge-bot[bot] committed:
peergos: 1.1.0 -> 1.2.0 (#402030)

Co-authored-by: christoph-heiss <contact@christoph-heiss.at>

Defelo committed:
gotosocial: move to codeberg (#402238)

nikstur committed:
repart: Enable discard option (#397906)

natsukium committed:
python312Packages.pyghmi: 1.5.77 -> 1.6.0 (#402364)

wolfgangwalther committed:
opensmtpd-extras: drop in favor of standalone filters (#376913)

nixpkgs-merge-bot[bot] committed:
syft: 1.22.0 -> 1.23.1 (#402439)

Co-authored-by: 06kellyjac <06kellyjac@users.noreply.github.com>

Atemu committed:
androidenv: support linking "latest" version of all plugins (fixing NDK linking) (#402391)

GaetanLepage committed:
zed-editor: 0.183.10 -> 0.183.11 (#402484)

msanft committed:
libnvidia-container: 1.17.2 -> 1.17.6 (#402488)

drupol committed:
blender: fix openusd support (#399716)

NickCao committed:
opencamlib: init at 2023.01.11, and use in freecad (#400063)

cardealerT committed:
maintainers: add cardealerT & A1egator

cardealerT committed:
kraken2: init at 2.1.5

cardealerT committed:
Merge branch 'master' of https://github.com/cardealerT/nixpkgs

cardealerT commented:
@acid-bong im really not smart and blew this up, if you still want to review I've made a new pull request. #402555

acid-bong commented:
Oh god, did you rebase or what?
